src/NHibernate/Persister/Collection/AbstractCollectionPersister.cs

```diff
@@ -1512,12 +1512,12 @@ private static void CalcPropertyColumnSpan(IType propertyType, ref int count)
 
 		public int GetSize(object key, ISessionImplementor session)
 		{
-			using(new SessionIdLoggingContext(session.SessionId))
+			using (session.BeginProcess())
 			try
 			{
 				if(session.EnabledFilters.Count > 0)
 				{
-					
+					// The above call validates the filters
```

Comment:
And it is also checking the status, but that is now already done due to `BeginProcess` above instead of just setting the id.